draft-ietf-privacypass-architecture.md

```diff
@@ -107,24 +82,72 @@ of all participating entities.
-each server; and, the number of supported ciphersuites. There are also
-client behavior patterns that can reduce the effective security of the
-server.
-
-In this document, we will provide a structural framework for building
-the ecosystem around the Privacy Pass protocol. The core of the document
-also includes policies for the following considerations.
+Privacy Pass is a protocol for authorization based on anonymous-credential
+authentication mechanisms. Typical approaches for authorizing clients,
+such as through the use of long-term cookies, are not privacy friendly
+since they allow servers to track clients across sessions and interactions.
+Privacy Pass takes a different approach: instead of presenting linkable
+state carrying information to servers, e.g., whether or not the client
+is an authorized user or has completed some prior challenge, clients
+present unlinkable proofs that attest to this information.
+
+The original Privacy Pass protocol provides a set of cross-origin
+authorization tokens that protect the client's anonymity during interactions
+with a server. This allows clients to communicate an attestation of a
+previously authenticated server action, without having to reauthenticate
+manually. The tokens retain anonymity in the sense that the act of
+revealing them cannot be linked back to the session where they were
+initially issued.
+
+At a high level, Privacy Pass is composed of two protocols: issuance
+and redemption. Issuance is a protocol between three logical entities:
+Client, Mediator, and Issuer. The Issuer is responsible for issuing
+tokens in response to requests from Clients. The Mediator is responsible
+for attesting properties about the Client for which tokens are issued.
+For example, in the original Privacy Pass protocol {{PPSRV}}, tokens were
+only issued to Clients that solved CAPTCHAs. In this context, the Mediator
+attested that some client solved a CAPTCHA and the resulting token
+produced by the Issuer was proof of this fact. Depending on the information
+being attested, Mediators may also store state about individual Clients,
+such as the number of overall tokens issued thus far.
+
+The redemption protocol runs between Client and Origin (server). It allows
+Origins to challenge Clients to present one or more tokens for authorization.
+Depending on the type of token, e.g., whether or not it is cross-origin
+or per-origin, and whether or not it can be cached, the Client either presents
+a previously obtained token or invokes the issuance protocol to acquire one
+for authorization.
+
+The issuance and redemption protocols operate in concert as shown in
+the figure below.
+
+~~~
+ Client        Mediator          Issuer          Origin
+
+    <---------------------------------------- Challenge \
+                                                        |
++--------------------------------------------\          |
+| TokenRequest --->                          |          |
```

Comment:
Perhaps the Client -> Mediator flow should indicate a first authentication/validation step?

Reply:
I like the later diagram that has "attest"